internal/security/acctz/acctz.go

```diff
@@ -0,0 +1,917 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package acctz provides helper APIs to simplify writing acctz test cases.
+package acctz
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/openconfig/gnmi/proto/gnmi"
+	"github.com/openconfig/gnoi/system"
+	"github.com/openconfig/gnsi/acctz"
+	authzpb "github.com/openconfig/gnsi/authz"
+	"github.com/openconfig/gnsi/credentialz"
+	gribi "github.com/openconfig/gribi/v1/proto/service"
+	tpb "github.com/openconfig/kne/proto/topo"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/binding"
+	"github.com/openconfig/ondatra/binding/introspect"
+	ondatragnmi "github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+	p4pb "github.com/p4lang/p4runtime/go/p4/v1"
+	"golang.org/x/crypto/ssh"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+const (
+	successUsername      = "acctztestuser"
+	successPassword      = "verysecurepassword"
+	failUsername         = "bilbo"
+	failPassword         = "baggins"
+	failRoleName         = "acctz-fp-test-fail"
+	successCliCommand    = "show version"
+	failCliCommand       = "show version"
+	shellCommand         = "uname -a"
+	gnmiCapabilitiesPath = "/gnmi.gNMI/Capabilities"
+	gnoiPingPath         = "/gnoi.system.System/Ping"
+	gnsiGetPath          = "/gnsi.authz.v1.Authz/Get"
+	defaultSSHPort       = 22
+)
+
+var gRPCClientAddr net.Addr
+
+// Record represents the structure for an acctz record.
+type Record struct {
```

Comment:
I think I asked about this in a different PR, but why do this? why not just use a []*pb.<type> ?

Using this new struct means you can't easily cmp.Diff() the expeted content and the received content for a proto... Something like:

```golang
got := fooReturnsPB()
if diff := cmp.Diff(got, want, cmpoptions.ProtoTransform()); diff != "" {
  t.Errorf("got/want mismatch, diff: %v", diff)
}
```

SHOULD do a comparison of the proto bits you got vs what  you wanted... and save what looks like ~100 lines of code in the tests where you pull apart all of the protos and records (and 50+ lines where you  fill record struct content...)

Reply:
Removed record struct & using pb directly for `cmp.Diff()`